tests/AuthService.test.js

```diff
@@ -10,19 +14,31 @@ describe ("AuthService", () => {
       : expect(() => {AuthService.encryptPassword(password)}).toThrow();
     }
   });
-  it("Token generate and check", () => {
+
+  it("Token generate and check", async () => {
     const operator = {
       id: 23,
       email: "example@gmail.com",
       passwordHash: AuthService.encryptPassword("heDK4src4w")
     };
     const token = AuthService.generateToken(operator);
+    const expectedOutput = Object.assign(
+      _.pick(operator, ...AuthService.OPERATOR_ATTRIBUTES_FOR_JWT),
+      {exp: jwt.verify(token, config.get("jwt.secret")).iat + config.get("jwt.lifetime")}
+    );
     const decodedToken = AuthService.checkAndDecodeToken(token);
-    expect(decodedToken.id).toBe(operator.id);
-    expect(decodedToken.email).toBe(operator.email);
-    expect(decodedToken.exp).toBeGreaterThan(Date.now()/1000);
+    expect(decodedToken).toStrictEqual(expectedOutput);
+    await delay(config.get("jwt.lifetime")*1000 + 100);
+    expect(() => {AuthService.checkAndDecodeToken(token)}).toThrow("jwt expired");
   });
+
   describe("Get operator by credentials", () => {
+    let yaxysBuffer;
+
+    beforeAll( () => {
+      yaxysBuffer = global.yaxys;
```

Comment:
You didn't get the whole point. Yes, you should store old version of `global.yaxys` here, but also you can write into the `global.yaxys` right here. Instead, you do it inside the cycle, which is not wise.